src/Transformations.jl

```diff
@@ -260,6 +261,64 @@ function transform!(tn::TensorNetwork, config::ColumnReduction)
     return tn
 end
 
+Base.@kwdef struct SplitSimplification <: Transformation
+    atol::Float64 = 1e-10  # A threshold for SVD rank determination
+end
+
+function transform!(tn::TensorNetwork, config::SplitSimplification)
+    done = false  # This flag will be set to true when no further changes are made
+
+    while !done
+        done = true  # Assume no changes will be made this pass
+
+        # Create a copy of the list of tensors so we can modify the original while iterating
+        tensor_list = copy(tensors(tn))
+
+        for tensor in tensor_list
+            inds = labels(tensor)
+
+            partitions = Iterators.flatten(combinations(inds, r) for r = 1:(length(inds)-1))
+
+            # Iterate over all possible bipartitions of the tensor's indices
+            for bipartition in partitions
+                left_inds = collect(bipartition)
+                right_inds = setdiff(inds, left_inds)
+
+                # Perform an SVD across the bipartition
+                u, s, v = svd(tensor; left_inds=left_inds)
+
+                # Get the singular values and decide the rank
+                singular_values = diag(s)
+                rank_s = sum(singular_values .> config.atol)
+
+                if rank_s < length(singular_values)
+                    # Remove unnecessary data in u, s, v
+                    u = view(u, labels(s)[1] => 1:rank_s)
+                    s = view(s, (idx -> idx => 1:rank_s).(labels(s))...)
+                    v = view(v, labels(s)[2] => 1:rank_s)
```

Comment:
Maybe is better not to get views? If we get views, the underlying array will be around and it will be bigger than this views.

Also,
```suggestion
                if rank_s < size(s,1)
                    # truncate data
                    u = view(u, labels(s)[1] => 1:rank_s)
                    s = view(s, (idx -> idx => 1:rank_s).(labels(s))...)
                    v = view(v, labels(s)[2] => 1:rank_s)
```